Improve access control documentation for remote exams (#15311)
PrairieLearn/PrairieLearn: docs/assessment/remoteExams.md

## Exams in a PrairieTest-managed testing center

For a PrairieTest-only access path, enable PrairieTest on the assessment and leave date control disabled.

In the UI:

1. Open the assessment **Access** tab.
2. Click **Edit** in **Defaults**.
3. Leave **Date control** disabled.
4. Under **Integrations**, enable **PrairieTest**.
5. Enter the PrairieTest exam UUID from the PrairieTest exam settings.
6. For **After completion**, choose what students can see after they finish while their reservation is still active.
7. Use the top-level **After completion** settings for what students can see after the reservation ends.
8. Click **Save**.

Some notes about this configuration:

- The PrairieTest exam UUID should be copied from PrairieTest for the specific exam. Each exam has its own unique UUID, and it's vital that the correct value is used for each separate exam.

> - Do not configure date control or a PrairieLearn time limit unless students should have a non-PrairieTest access path. PrairieTest enforces scheduling and time limits, including alternate exam windows and disability accommodations.

Reviewer: I'm thinking we can be more prescriptive here, given that these are just example configurations.

```suggestion
- Do not configure date control or a PrairieLearn time limit.
- Apply overrides and disability accommodations in PrairieTest. Do not use overrides in PrairieLearn.
```

7. Under **After completion**, set **Question visibility** to **Hide questions permanently**. Leave **Score visibility** visible after completion if students should see their total score.
8. For students with extra-time accommodations, click **Add override**. Choose **Students by label** if you already maintain an accommodation label such as "Extended time"; otherwise choose **Specific students**.
9. In the extra-time override, click **Override** next to **Due date** and **Time limit**. For 1.5x time on a 60-minute exam, use a 90-minute time limit and extend the due date to cover that time.
10. For students taking the exam in an alternate window, click **Add override**. Choose **Specific students** unless this is a reusable cohort that should be managed by label.
11. In the alternate-window override, click **Override** next to **Release**, **Due date**, and **Time limit** for the alternate exam window.
12. For students who need both an alternate window and extra time, add a student-specific override or a dedicated reusable-cohort label override listed below the other matching overrides.
13. Click **Save**.

Some notes about this configuration:

> - The exam window has been set to be 5 minutes longer than the exam time limit. However, students will not be able to submit past the due time under any circumstances. If a student starts this exam more than 5 minutes late, then the countdown timer on their exam will reflect the time remaining until the due time.

Reviewer: Because the example doesn't actually appear until the JSON below, I think it's better to word these instructions declaratively rather than descriptively.

```suggestion
- Set the exam window (release to due date) to be 5 minutes longer than the exam time limit. Students will not be able to submit past the due date under any circumstances. If a student starts this exam more than 5 minutes late, then the countdown timer on their exam will reflect the time remaining until the due date.
```

- If a student closes their web browser accidentally during an exam, they can reopen PrairieLearn and continue where they left off. They can also switch computers and continue the same exam. The timer does not pause while the browser is closed.
- Remember to extend both the due date and time limit for students with extra-time accommodations.
- After the timer expires the exam will auto-close and grade any saved but ungraded questions. Students cannot submit after their timer expires or after the due time, whichever comes first. Once the assessment is complete, students can see their final score but cannot review any questions.
- If a student closes their web browser before the exam is complete, their exam will be automatically closed and graded within 12 minutes after their timer expires. If they try and access their exam during this time it will immediately close and grade.
- Before downloading final scores, wait at least 12 minutes after the last student would have finished to ensure all exams are closed. You can also check and manually close exams on the **Students** page under the assessment in PrairieLearn.

> - Because no after-deadline submission mode is configured, submissions are not allowed after the due time. The **Question visibility** setting keeps completed exam questions hidden while the total score remains visible by default. This does not prevent students from seeing questions or grading feedback while they are taking the exam.

Reviewer: ```suggestion
- Because no after-deadline submission mode is configured, submissions are not allowed after the due date. The **Question visibility** setting keeps completed exam questions hidden while the total score remains visible by default. This does not prevent students from seeing questions or grading feedback while they are taking the exam.
```

??? info "JSON"

    This reference shows an extra-time label override and an alternate-window override for specific students.

    ```json title="infoAssessment.json"
    {
      "accessControl": [
        {
          "dateControl": {
            "release": { "date": "2020-04-20T11:00:00" },
            "due": { "date": "2020-04-20T12:05:00" },
            "durationMinutes": 60
          },
          "afterComplete": {
            "questions": { "hidden": true }
          }
        },
        {
          "uuid": "22222222-2222-4222-8222-222222222222",
          "labels": ["Extended time"],
          "dateControl": {
            "due": { "date": "2020-04-20T12:40:00" },
            "durationMinutes": 90
          }
        },
        {
          "uuid": "33333333-3333-4333-8333-333333333333",
          "dateControl": {
            "release": { "date": "2020-04-20T23:00:00" },
            "due": { "date": "2020-04-21T00:05:00" },
            "durationMinutes": 60
          }
        }
      ]
    }
    ```

## Asynchronous, timed exams

> We do **NOT** recommend exams to be run using this configuration for high-stakes exams. While giving exams asynchronously will simplify exam administration and provide students with more flexibility, it comes at the expense of making it easier to cheat. We recommend [synchronous, timed exams](#synchronous-timed-exams).

Reviewer: ```suggestion
We do **NOT** recommend running high-stakes exams using this configuration. While giving exams asynchronously will simplify exam administration and provide students with more flexibility, it comes at the expense of making it easier to cheat. We recommend [synchronous, timed exams](#synchronous-timed-exams) unless you have a secure testing environment.
```

In the UI:

1. On the assessment **Settings** page, under **Grading**, clear **Allow real-time grading** and save the settings.
2. On the assessment **Access** page, click **Edit** in **Defaults**.
3. Enable **Date control**.
4. Set **Release** to the exam start time, such as Apr 20 at 11:00am.
5. Set **Due date** to the exam end time, such as Apr 20 at 12:10pm.
6. Enable **Time limit** and enter the ordinary working time, such as 60 minutes.
7. Under **After completion**, set **Question visibility** to **Hide questions permanently**.
8. For students with extra-time accommodations, click **Add override**. Choose **Students by label** if you already maintain an accommodation label such as "Extended time"; otherwise choose **Specific students**.
9. In the extra-time override, click **Override** next to **Due date** and **Time limit**. For 1.5x time on a 60-minute exam, use a 90-minute time limit and extend the due date to cover that time.
10. For students taking the exam in an alternate window, click **Add override**. Choose **Specific students** unless this is a reusable cohort that should be managed by label.

> 11. In the alternate-window override, click **Override** next to **Release**, **Due date**, and **Time limit** for the alternate exam window.

Author: Codex repeated a lot of this information across each section to mirror the original JSON. This is probably fine? I'd be happy treating this PR as a step in the right direction and leaving further improvements on the table. The main goal is to make this non-JSON-first.

Reviewer: I'm fine with the repetition for now.
